independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/RequestContext.java

```diff
@@ -176,4 +177,24 @@ private void fireIfNotEmpty(LazyValue<Notifier<Object>> value) {
                 ArcContainerImpl.instance());
     }
 
+    class RequestContextState implements ContextState {
+
+        private final ConcurrentMap<Contextual<?>, ContextInstanceHandle<?>> value;
+
+        RequestContextState(ConcurrentMap<Contextual<?>, ContextInstanceHandle<?>> value) {
+            this.value = value;
+        }
+
+        @Override
+        public Map<InjectableBean<?>, Object> getContextualInstances() {
+            return value.values().stream()
+                    .collect(Collectors.toMap(ContextInstanceHandle::getBean, ContextInstanceHandle::get));
+        }
+
+        ConcurrentMap<Contextual<?>, ContextInstanceHandle<?>> getConcurrentMap() {
```

Comment:
I think that we could just make the `RequestContextState.value` private.

Reply:
Fair enough